core/src/main/java/org/ehcache/core/internal/util/CollectionUtil.java

```diff
@@ -50,4 +52,78 @@ public static int findBestCollectionSize(Iterable<?> iterable, int bestBet) {
   public static <K, V> Map.Entry<K, V> entry(K key, V value) {
     return new AbstractMap.SimpleImmutableEntry<>(key, value);
   }
+
+  /**
+   * Copy each map entry to a new map but check that each key and value isn't null. Throw
+   * a {@code NullPointerException} if it's the case.
+   *
+   * @param entries entries to copy
+   * @param <K> type of key
+   * @param <V> type of value
+   * @return cloned map
+   * @throws NullPointerException if a key or value is null
+   */
+  public static <K, V> Map<K, V> copyMapButFailOnNull(Map<? extends K, ? extends V> entries) {
+    Map<K, V> entriesToRemap = new HashMap<>(entries.size());
+    entries.forEach((k, v) -> {
+      // If a key/value is null, throw NPE, nothing gets mutated
+      if (k == null || v == null) {
+        throw new NullPointerException();
+      }
+      entriesToRemap.put(k, v);
+    });
+    return entriesToRemap;
+  }
+
+  /**
+   * Create a map with one key and a corresponding value.
+   *
+   * @param key0 the key
+   * @param value0 the value
+   * @param <K> key type
+   * @param <V> value type
+   * @return map with one entry
+   */
+  public static <K, V> Map<K, V> map(K key0, V value0) {
```

Comment:
Why not just inline this?

Reply:
On second thoughts... looks like these method are only used in unit tests for this class?  Are they needed?

Reply:
I use them in tests indeed. The singleton one is there for completude. I didn't one to add a new CollectionUtil for test only. And they are in Java 9 JDK. I don't think it's harmful to keep them there.